packages/component-meta/lib/base.ts

```diff
@@ -297,6 +306,36 @@ ${commandLine.vueOptions.target < 3 ? vue2TypeHelpersCode : typeHelpersCode}
 			return 0;
 		}
 
+
+		function getExtendedMeta() {
+			let meta: Record<string, any> = {};
+			const snapshot = language.scripts.get(componentPath)?.snapshot!;
+			const fileText = snapshot.getText(0, snapshot.getLength());
+			const ast = ts.createSourceFile('/temp', fileText, ts.ScriptTarget.Latest, true);
+			const identifier = 'defineComponentMeta'
+			const printer = ts.createPrinter(checkerOptions.printer);
+
+			if (!ast.identifiers.get(identifier)) return
+
+			function traverse(node: ts.Node): void {
+				if (ts.isIdentifier(node) && node.text === identifier) {
+					const argument = node.parent.arguments[0];
+					if (ts.isObjectLiteralExpression(argument)) {
+						argument.properties.forEach(property => {
+							if (!ts.isPropertyAssignment(property)) return
+							const key = property.name.getText();
+							const valueNode = resolveDefaultOptionExpression(property.initializer, ts);
+							meta[key] = printer?.printNode(ts.EmitHint.Expression, valueNode, ast) ?? valueNode.getText(ast)
```

Comment:
@KazariEX I'm wondering if it's a good idea to parse nested objects and arrays by traversing the rest of the AST here. I tried to match the parser for prop default values.

On one side it would be nice for end users to receive plain objects / arrays, but might be a bit heavy on performances.